engine/swarm/services.md

```diff
@@ -70,6 +70,42 @@ $ docker service create --name helloworld alpine ping docker.com
 9uk4639qpg7npwf3fn2aasksr
 ```
 
+### Specify the image version the service should use
+
+When you create a service without specifying any details about the version of
+the image to use, the service uses the latest version available in Docker Cloud
+or your local registry. You can force the service to use a specific version of
+the image in a few different ways, depending on your desired outcome.
+
+When you don't specify a tag, you get the latest version of an image, because
+Docker appends the `latest` tag. The following two commands are equivalent:
+
+```bash
+$ docker service create ubuntu
+
+$ docker service create ubuntu:latest
+```
+
+You can specify any tag which has been pushed to Docker Hub or your private
+registry. For instance, some of the tags available for the `ubuntu` image are
+`latest`, `devel`, `yakkety`, and `16.04`.
+
+### Update a service's image after creation
+
+Each tag represents a digest, similar to a Git hash. Some tags, such as
+`latest`, are updated occasionally to point to a new digest. Services are
+constrained to run a specific digest of an image until you update the service
+using `service update` with the `-image` flag, which is supported in Docker
+Engine 1.13 and higher. If you use an older version of Docker Engine, you must
+remove and re-create the service to update its image.
+
+When you run `service update` with the `-image` flag, a background function
+called `ResolveTagToDigest` queries Docker Hub or your private Docker registry
+for the digest the tag points to, and updates the service tasks to use that
+digest. If the tag does not exist or the registry is not reachable, the service
```

Comment:
We should not mention the name of the function from our source code.

```
When you create a service, or run `service update` and specify a new image with
the `--image` flag, Docker queries Docker Hub or your private Docker registry for
the digest the tag points to. It stores that digest on the service so that tasks will use
the image with this digest, ensuring that each node running the service uses the
same version of the image, even if the tag is updated while the service is running.
```